### ISSUE TITLE
ENH: Add automated bad-channel marking

### ISSUE THREAD

**Issue by larsoner** [mne-tools/mne-nirs#278]
I recently migrated a pipeline from HOMER to MNE-Python. The one function I was missing was one that marked a channel in a recording if the optical density was over 3 standard deviations from the group mean (I think this was the criteria anyway, and it seems to match what they suggest offhand in [their publication](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2761652/#S11title)). Is it worth implementing something like this here? If so I have a bunch of recordings I can validate against. If not, I'm curious to know what you use instead @rob-luke !

It's not 100% clear to me, but I could see this being implemented as either 1) 3 std devs away from the mean *across all channels*, or 2) operating based on the std/mean of *each channel* independently -- probably the latter makes more sense. Then we'd want to use channel-specific annotations from MNE to mark segments as bad for segments of time during which it exceeded the threshold. Then users can choose to mark the entire channel bad for a given file/run (which the previous analysis I have access to did) or to have those segments marked bad in time. However, this latter method will be a bit tricky because in principle I think we'd want to exclude that interval from the GLM somehow -- which seems hard to do for the entire recording, let alone just for a single channel.

Mostly thinking out loud here @rob-luke, perhaps it makes sense to have a quick chat about this at some point!

**Comment by rob-luke** [comment]
I've always found that the scalp coupling index identifies bad channels quite well, and so I didn't need a std based metric. Have you found cases where SCI doesn't mark the channels you want?

https://mne.tools/stable/generated/mne.preprocessing.nirs.scalp_coupling_index.html?highlight=coupling#mne.preprocessing.nirs.scalp_coupling_index

I have also been wanting to implement the peak spectral power metric for identifying bad segments of the recording (whereas sci finds bad channels for the entire recording). As SCI can incorrectly mark a channel as good if it has lots of artifacts. I think this might be a better method than the moving std approach

See https://github.com/lpollonini/qt-nirs and 

Hernandez, Samuel Montero, and Luca Pollonini. "NIRSplot: a tool for quality assessment of fNIRS scans." Optics and the Brain. Optical Society of America, 2020.

**Comment by rob-luke** [comment]
This is not to say we shouldn't add the std metric. I'm happy with that, more options the better. I just haven't had need for it. And would suggest the PSP method is a more considered approach.

**Comment by larsoner** [comment]
After a quick chat @rob-luke and I settled on using peak spectral power (PSP) to do channel-wise annotations of bad segments, then use the scalp coupling index (SCI) computed using only good segments only. Sounds like @rob-luke has some existing MATLAB code that can be adapted with permission under our BSD 3-clause license

TL;DR: Not adding std-based bad channel marking, instead improving SCI using PSP

**Comment by rob-luke** [comment]
A few notes for myself...

The output should be able to generate a nice figure like: https://mne.tools/mne-bids-pipeline/examples/ds003392/sub-01_task-localizer_report.html#11

And the naming of arguments should be similar to https://github.com/mne-tools/mne-python/blob/f1e6097c36e7232514d67d4b04af568e36ac5758/mne/preprocessing/maxwell.py#L1927
